### PR TITLE
feature/39520

### PR DESCRIPTION
**Pull request by abhipunj0812**
Code: TypeScript Implementation

### CODE REVIEW

File: extraTs/patterns/diamond.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+export function createDiamondShape(rows) {
```
Comment (rahulsainidev) [inline extraTs/patterns/diamond.ts]: Add argument type and return type to function

File: extraTs/patterns/equilateral.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+ export function createEquilateral(rows) {
```
Comment (rahulsainidev) [inline extraTs/patterns/equilateral.ts]: Add argument type and return type

File: extraTs/utils/helper.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export function validateEmail(email) {
```
Comment (rahulsainidev) [inline extraTs/utils/helper.ts]: Add argument and return type

File: extraTs/utils/permissions.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+// tslint:disable-next-line:one-line
+export default function hasPermission(moduleName, role, permissionType){
```
Comment (rahulsainidev) [inline extraTs/utils/permissions.ts]: Add argument and return type

File: extraTs/utils/permissions.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    let type;
+    const { all, read, write, Delete } = moduleName;
+    // tslint:disable-next-line: triple-equals
```
Comment (rahulsainidev) [inline extraTs/utils/permissions.ts]: Do not use disable linting, just fix the types and warnings as per lint rules.

**Review by rahulsainidev — CHANGES_REQUESTED** [review]

File: extraTs/patterns/diamond.ts (hunk 1 above)
Comment (abhipunj0812) [reply extraTs/patterns/diamond.ts]: ok sir

**Review by rahulsainidev — APPROVED** [review]

**Pull request merged**